VIPERBuilder/Classes/VIPERBuilder.swift

```diff
@@ -22,11 +22,17 @@
 import UIKit
 
 open class VIPERBuilder<I: VIPERInteractor, P: VIPERPresenter, R: VIPERRouter>: NSObject {
-    public let interactor = I()
-    public let presenter = P()
-    public let router = R()
+    public let interactor: I
+    public let presenter: P
+    public let router: R
     
-    public init(controller: UIViewController) {
+    public init(interactor: I = I(),
+                presenter: P = P(),
+                router: R = R(),
+                controller: UIViewController) {
```

Comment:
I think Swift best practice is to have the parameters with default values be at the end so that the method signature is more predictable when including non-default values.

Reply:
Great point, will address